internal/datasources/service/convert.go

```diff
@@ -0,0 +1,62 @@
+// SPDX-FileCopyrightText: Copyright 2024 The Minder Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package service
+
+import (
+	"errors"
+	"fmt"
+
+	"google.golang.org/protobuf/encoding/protojson"
+
+	"github.com/mindersec/minder/internal/datasources"
+	"github.com/mindersec/minder/internal/db"
+	minderv1 "github.com/mindersec/minder/pkg/api/protobuf/go/minder/v1"
+)
+
+func dataSourceDBToProtobuf(ds db.DataSource, dsfuncs []db.DataSourcesFunction) (*minderv1.DataSource, error) {
+	outds := &minderv1.DataSource{
+		Version: minderv1.VersionV1,
+		Type:    string(minderv1.DataSourceResource),
+		Id:      ds.ID.String(),
+		Name:    ds.Name,
+		Context: &minderv1.ContextV2{
+			ProjectId: ds.ProjectID.String(),
+		},
+	}
+
+	if len(dsfuncs) == 0 {
+		return nil, errors.New("data source is invalid and has no defintions")
+	}
+
+	// All data source types should be equal... so we'll just take the first one.
+	dsfType := dsfuncs[0].Type
```

Comment:
Should we ensure/validate that all sources have the same type?

Reply:
We should. I was thinking of doing this on the `create` and `update` parts though. There's not much we can do on the `get` side.